core/src/main/java/com/redhat/lightblue/client/response/LightblueException.java

```diff
@@ -0,0 +1,257 @@
+    public static final String ERR_AUDIT_HOOK_MISSING_ID = "audit-hook:MissingID";
+    public static final String ERR_REST_CRUD_REST_ERROR = "rest-crud:RestError";
+    public static final String ERR_REST_CRUD_REST_FIND = "rest-crud:RestFindError";
+    public static final String ERR_REST_CRUD_REST_INSERT = "rest-crud:RestInsertError";
+    public static final String ERR_REST_CRUD_REST_UPDATE = "rest-crud:RestUpdateError";
+    public static final String ERR_REST_CRUD_REST_SAVE = "rest-crud:RestSaveError";
+    public static final String ERR_REST_CRUD_REST_DELETE = "rest-crud:RestDeleteError";
+    public static final String ERR_REST_CRUD_NO_ENTITY_MATCH = "rest-crud:NoEntityMatch";
+    public static final String ERR_REST_CRUD_NO_VERSION_MATCH = "rest-crud:NoVersionMatch";
+    public static final String ERR_REST_CRUD_CANT_GET_MEDIATOR = "rest-crud:CantGetMediator";
+    public static final String ERR_REST_CRUD_CANT_GET_TRANSLATOR = "rest-crud:CantGetTranslator";
+    public static final String ERR_REST_METADATA_REST_ERROR = "rest-metadata:RestError";
+    public static final String ERR_REST_METADATA_NO_ENTITY_NAME = "rest-metadata:NoEntityName";
+    public static final String ERR_REST_METADATA_NO_ENTITY_VERSION = "rest-metadata:NoEntityVersion";
+    public static final String ERR_REST_METADATA_NO_ENTITY_STATUS = "rest-metadata:NoEntityStatus";
+    public static final String ERR_REST_METADATA_NO_NAME_MATCH = "rest-metadata:NoNameMatch";
+    public static final String ERR_REST_METADATA_NO_VERSION_MATCH = "rest-metadata:NoVersionMatch";
+    public static final String ERR_REST_METADATA_CANT_GET_METADATA = "rest-metadata:CantGetMetadata";
+    public static final String ERR_REST_METADATA_CANT_GET_PARSER = "rest-metadata:CantGetParser";
+    public static final String ERR_REST_METADATA_CANT_GET_TRANSLATOR = "rest-metadata:CantGetJsonTranslator";
+
+    private DefaultLightblueResponse lightblueResponse;
+    private Set<String> errorCodes = new HashSet<>();
+
+
+    public LightblueException() {
+        super();
+    }
+
+    public LightblueException(String message, DefaultLightblueResponse lightblueResponse) {
+        super(message);
+        this.lightblueResponse = lightblueResponse;
+        extractErrorCodes();
+    }
+
+    public LightblueException (String message, DefaultLightblueResponse lightblueResponse, Throwable cause) {
+        super (message,cause);
+        this.lightblueResponse = lightblueResponse;
+        extractErrorCodes();
+    }
+
+    public DefaultLightblueResponse getLightblueResponse() {
+        return lightblueResponse;
+    }
+
+    public void setLightblueResponse(DefaultLightblueResponse lightblueResponse) {
+        this.lightblueResponse = lightblueResponse;
+    }
+
+    public boolean exists(String errorCode) {
+        return errorCodes.contains(errorCode);
+    }
+
+    public Set<String> getErrorCodes() {
+        return errorCodes;
```

Comment:
You could do getErrorCodes().contains(errorCode)

Not a big deal though.

On Fri, Jul 10, 2015 at 8:51 AM, Yusuf Kör <notifications@github.com> wrote:

> In
> core/src/main/java/com/redhat/lightblue/client/response/LightblueException.java
> <https://github.com/lightblue-platform/lightblue-client/pull/132#discussion_r34361464>
> :
>
> > +    }
> > +
> > +    public DefaultLightblueResponse getLightblueResponse() {
> > +        return lightblueResponse;
> > +    }
> > +
> > +    public void setLightblueResponse(DefaultLightblueResponse lightblueResponse) {
> > +        this.lightblueResponse = lightblueResponse;
> > +    }
> > +
> > +    public boolean exists(String errorCode) {
> > +        return errorCodes.contains(errorCode);
> > +    }
> > +
> > +    public Set<String> getErrorCodes() {
> > +        return errorCodes;
>
> I would then have to initialize the error codes in "exists" method as
> well, which I don't like.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/132/files#r34361464>
> .
>